### PR TITLE
Master stock pim

### PR DESCRIPTION
**Pull request by pimodoo**

### CODE REVIEW

File: addons/delivery/wizard/choose_delivery_package.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            res = stock_quant_package.packaging_id
+        return res
+
+    def _default_shipping_weight(self):
+        if self.env.context.get('default_stock_quant_package_id'):
+            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
+            return stock_quant_package.shipping_weight
+        else:
+            picking_id = self.env['stock.picking'].browse(self.env.context['active_id'])
+            pack_operation_ids = [po for po in picking_id.pack_operation_ids if po.qty_done > 0 and not po.result_package_id]
```
Comment (jco-odoo) [inline addons/delivery/wizard/choose_delivery_package.py]: maybe check entire packages also: not x.product_id -> x.package_id.shipping_weight/weight

File: addons/delivery/views/delivery_view.xml [hunk 2]
```diff
@@ -4,6 +4,14 @@
         <menuitem id="menu_delivery" name="Delivery" parent="stock.menu_stock_config_settings" groups="stock.group_stock_manager" sequence="50"/>
         <menuitem id="sale_menu_delivery" name="Delivery" parent="sales_team.menu_sale_config" groups="sales_team.group_sale_manager" sequence="12"/>
 
+        <record model="ir.actions.act_window" id="action_delivery_packaging_view">
```
Comment (dbo-odoo) [inline addons/delivery/views/delivery_view.xml]: ideally, `id` should come before `model`

File: addons/delivery/wizard/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+import choose_delivery_package
```
Comment (dbo-odoo) [inline addons/delivery/wizard/__init__.py]: coding utf-8 blablabl

File: addons/delivery/wizard/choose_delivery_package.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class ChooseDeliveryPackage(models.TransientModel):
+    _name = 'choose.delivery.package'
+    _description = 'Choose delivery package type when there is a carrier on the picking'
```
Comment (dbo-odoo) [inline addons/delivery/wizard/choose_delivery_package.py]: `_description` will be the name of the model in the Settings > Technical > Models list view, I suggest something more name-like ('Delivery Package Selection Wizard' or some other crap of the same kind)

File: addons/delivery/wizard/choose_delivery_package.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    stock_quant_package_id = fields.Many2one(
+        'stock.quant.package',
+        default=lambda self: self._default_stock_quant_package_id()
+    ) 
+    delivery_packaging_id = fields.Many2one(
+        'product.packaging',
+        domain=lambda self: self._domain_delivery_packaging_id(),
```
Comment (dbo-odoo) [inline addons/delivery/wizard/choose_delivery_package.py]: should be passed by reference (without the `()`) or it's only evaluated at server startup I think (I had problems with that in the pas)

File: addons/delivery/wizard/choose_delivery_package.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    shipping_weight = fields.Float(
+        string='Shipping Weight',
+        default=lambda self: self._default_shipping_weight()
+    )
+
+    def _default_stock_quant_package_id(self):
+        if self.env.context.get('default_stock_quant_package_id'):
+            return self.env['stock.quant.package'].browse(self.env.context['stock_quant_package_id'])
+
+    def _domain_delivery_packaging_id(self):
```
Comment (dbo-odoo) [inline addons/delivery/wizard/choose_delivery_package.py]: should be `@api.model` (like all your default functions as well)

File: addons/delivery/wizard/choose_delivery_package.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        if self.env.context.get('default_delivery_packaging_id'):
+            res = self.env['product.packaging'].browse(self.env.context['default_delivery_packaging_id'])
+        if self.env.context.get('default_stock_quant_package_id'):
+            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
+            res = stock_quant_package.packaging_id
+        return res
+
+    def _default_shipping_weight(self):
+        if self.env.context.get('default_stock_quant_package_id'):
+            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
```
Comment (dbo-odoo) [inline addons/delivery/wizard/choose_delivery_package.py]: intermediate variables can make your code more readable

File: addons/delivery/wizard/choose_delivery_package_views.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="choose_delivery_package" model="ir.ui.view">
```
Comment (dbo-odoo) [inline addons/delivery/wizard/choose_delivery_package_views.xml]: `id="choose_delivery_package_view_form"`
`<field name="name">choose.delivery.package.form</field>`

**Pull request merged**